src/use-built-ins-plugin.js

```diff
@@ -0,0 +1,227 @@
+import { definitions } from "./built-in-definitions";
+
+function isPolyfillSource(value) {
+  return value === "babel-polyfill";
+}
+
+function warnOnInstanceMethod(details) {
+  console.warn(
+    `Adding a polyfill: An instance method may have been used: ${details}`,
+  );
+}
+
+function has(obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key);
+}
+
+function getObjectString(node) {
+  if (node.type === "Identifier") {
+    return node.name;
+  } else if (node.type === "MemberExpression") {
+    return `${getObjectString(node.object)}.${getObjectString(node.property)}`;
+  } else {
+    return "";
+  }
+}
+
+export default function({ types: t }) {
+  function addImport(path, builtIn, builtIns) {
+    if (builtIn && !builtIns.has(builtIn)) {
+      builtIns.add(builtIn);
+      const importDec = t.importDeclaration([], t.stringLiteral(builtIn));
+      importDec._blockHoist = 3;
+      const programPath = path.find(path => path.isProgram());
+      programPath.unshiftContainer("body", importDec);
+    }
+  }
+
+  function addUnsupported(path, polyfills, builtIn, builtIns) {
+    if (Array.isArray(builtIn)) {
+      for (const i of builtIn) {
+        if (polyfills.indexOf(i) !== -1) {
+          addImport(path, `core-js/modules/${i}`, builtIns);
+        }
+      }
+    } else {
+      if (polyfills.indexOf(builtIn) !== -1) {
+        addImport(path, `core-js/modules/${builtIn}`, builtIns);
+      }
+    }
+  }
+
+  function isRequire(path) {
+    return t.isExpressionStatement(path.node) &&
+      t.isCallExpression(path.node.expression) &&
+      t.isIdentifier(path.node.expression.callee) &&
+      path.node.expression.callee.name === "require" &&
+      path.node.expression.arguments.length === 1 &&
+      t.isStringLiteral(path.node.expression.arguments[0]) &&
+      isPolyfillSource(path.node.expression.arguments[0].value);
+  }
+
+  const addAndRemovePolyfillImports = {
+    ImportDeclaration(path) {
+      if (
+        path.node.specifiers.length === 0 &&
+        isPolyfillSource(path.node.source.value)
+      ) {
+        console.warn(
+          `
+Adding "import 'babel-polyfill'" isn't necessary with the useBuiltIns option anymore.
+Please remove the call.
+`,
+        );
+        path.remove();
+      }
+    },
+    Program: {
+      enter(path, state) {
+        if (!state.opts.polyfills) {
+          throw path.buildCodeFrameError(
+            `
+There was an issue in "babel-preset-env" such that
+the "polyfills" option was not correctly passed
+to the "transform-polyfill-require" plugin
+`,
+          );
+        }
+        path.get("body").forEach(bodyPath => {
+          if (isRequire(bodyPath)) {
+            console.warn(
+              `
+Adding "require('babel-polyfill')" isn't necessary with the useBuiltIns option anymore.
```

Comment:
Avoid "anymore". This message is valid for new and old users. To users it *never* was necessary